### DISCUSSION TITLE
control diagnostic output with command line switches

### DISCUSSION THREAD

**Discussion by davetapley** [microsoft/pyright#5616] [Ideas]
Would you consider adding e.g. `--reportUnusedImport` and `--no-reportUnusedImport` to [the command line options](https://github.com/microsoft/pyright/blob/main/docs/command-line.md)?

Same reasoning as:
- https://github.com/microsoft/pyright/discussions/4491

**Comment by erictraut** [comment]
No, I don't think this could be justified. These are two of many dozens of fine-grained configuration options supported by pyright. If you want to specify fine-grained options like this, you should use a configuration file or file-level `# pyright` comments.

**Reply by davetapley** [reply]
Sorry, I should have been more clear: I'm proposing adding [all the options](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#type-check-diagnostics-settings), as switches, not just the two I gave as an example.

Similar to #4491, as @Pyprohly suggested there, this is a use case where it's not a 'permanent' change (where file based configuration is appropriate), but more for 'one off' tasks.

But, happy to wait and see if other interest, as with #4491
